src/tests/ftest/util/server_utils.py

```diff
@@ -48,6 +53,193 @@ class ServerFailed(Exception):
+        okay, failed, path = node_setup_okay(self.hosts, NodeListType.SERVER)
+        if not okay:
+            raise ServerFailed(
+                "Server node {} does not have directory {} set up correctly "
+                "for user {}.".format(failed, path, getpass.getuser()))
+
+        # Create the hostfile
+        self.hostfile = write_host_file(self.hosts, path, slots)
+
+    def update_configuration(self, basepath):
+        """Update the config parameter with a yaml file.
+
+        Args:
+            basepath (str): DAOS install basepath
+        """
+        self.config.update(create_server_yaml(basepath), "server.config")
+
+    def get_param_names(self):
+        """Get a sorted list of daos_server command parameter names."""
+        names = self.get_attribute_names(FormattedParameter)
+        names.extend(["request", "action"])
+        return names
+
+    def get_params(self, test, path="/run/daos_server/*"):
+        """Get values for all of the server command params using a yaml file.
+
+        Sets each BasicParameter object's value to the yaml key that matches
+        the assigned name of the BasicParameter object in this class. For
+        example, the self.block_size.value will be set to the value in the yaml
+        file with the key 'block_size'.
+
+        Args:
+            test (Test): avocado Test object
+            path (str, optional): yaml namespace.
+                Defaults to "/run/daos_server/*".
+
+        """
+        super(ServerCommand, self).get_params(test, path)
+
+    def get_launch_command(self, manager, uri=None, env=None):
+        """Get the process launch command used to run daos_server.
+
+        Args:
+            manager (str): mpi job manager command
+            uri (str, optional): path to uri file. Defaults to None.
+            env (dict, optional): dictionary on env variable names and values.
+                Defaults to None.
+
+        Raises:
+            ServerFailed: if the specified job manager is unsupported.
+
+        Returns:
+            str: daos_sever launch command
+
+        """
+        if manager is None:
+            # Run locally
+            return self.__str__()
+
+        elif manager.endswith("orterun"):
+            if env:
+                assign_env = [
+                    "{}={}".format(key, val) for key, val in env.items()]
+            args = [
+                "-np {}".format(len(self.hosts)),
+                "-hostfile {}".format(self.hostfile),
+                "--enable-recovery",
+            ]
+            args.extend(["-x {}".format(assign) for assign in assign_env])
+            if uri is not None:
+                args.append("--report-uri {}".format(uri))
+
+        else:
+            raise ServerFailed("Unsupported job manager: {}".format(manager))
+
+        return "{} {} {}".format(manager, " ".join(args), self.__str__())
+
+    def start(self, manager, verbose=True, env=None, timeout=600):
```

Comment:
is there not a another similar method somewhere in the ftest framework that does a similar thing? or do we need this because it does something specifically different?

Reply:
Yes, we have the run_server function. Implementing the `ServerCommand.start()` is an effort to restructure the code, the goal is to remove the functions and create `ServerCommand` class methods to handle things like `start()` and `stop()`